server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -193,6 +201,9 @@ public void parse(DocumentParserContext context) throws IOException {
                     );
                 }
             }
+            if (wroteField) {
```

Comment:
I think we don't need  the `wroteField` variable?  If we are at this place, it means that the field contains non-empty value and without exceptions.

Reply:
Tbh, I think that we'd still reach this point **iff** the value passed was sth like: `embedding_vec: {}`. E.g. when running the following
```
PUT/my_index?pretty{"settings":{"number_of_shards":1,"number_of_replicas":0},"mappings":{"properties":{"name":{"type":"text"},"embedding_vec":{"type":"sparse_vector"}}}}
```
```
POST/my_index/_doc?pretty{"name":"JohnDoe","embedding_vec":{}}
```
```
POST/my_index/_doc?pretty{"name":"JaneDoe","embedding_vec":{"1":0.5,"5":0.5,"100":1}}
```
```
GET/my_index/_search?pretty{"query":{"exists":{"field":"embedding_vec"}}}
```
we get back both results. Not sure what the expectation is here however, and whether:
i) this is something that could actually occur
ii) we want to include **John** in the results or not. 

Personally, I'd opt for leaving him out - hence the check if if we have at least one value populated, but please do let me know if this is not the expected behaviour (or if I have missed something in the code/examples above).

Reply:
An empty sparse_vector still means that the field exists.

Think around `"text_field": ""`, that field is empty, but present.

We should allow empty to be considered present.

Reply:
However, I do NOT think "exists" should work for "null"

Reply:
`if (context.parser().currentToken() != Token.START_OBJECT) {` I think this prevents `sparse_vector: null`. 

So, we will always get at least `sparse_vector: {}`

Reply:
Hmm I see. I was confused a bit in what we would consider `null` for `sparse_vector`, but looking also at the docs it's just the empty array that's defined. And regardless, since it passes the initial checks w/o any exceptions and makes it to the index, it makes sense to also match an `exists` query.

To be honest, the difference in how we parse & translate the following wasn't clear to me (esp. because `sparse_vector` is a single valued field by design), but it makes sense now
```
PUT /my_index?pretty{"settings":{"number_of_shards":1,"number_of_replicas":0},"mappings":{"properties":{"embedding_vec":{"type":"sparse_vector"}}}}
```
```
POST /my_index/_doc?pretty{"embedding_vec":{}}
```
```
POST/my_index/_doc?pretty{"embedding_vec":[]}
```
```
GET /my_index/_search?pretty{"query":{"exists":{"field": "embedding_vec"}}}
```

Thanks for the comment & clarifications @mayya-sharipova & @benwtrent ! 🙏  Will update as suggested.